main.tf

```diff
@@ -0,0 +1,186 @@
+data "aws_region" "current" {}
+
+locals {
+  architecture_layer_suffix_map = {
+    x86_64 = "",
+    arm64  = "-ARM"
+  }
+  runtime_base = regex("[a-z]+", var.runtime)
+  runtime_base_environment_variable_map = {
+    nodejs = {
+      DD_LAMBDA_HANDLER = var.handler
+    }
+    python = {
+      DD_LAMBDA_HANDLER = var.handler
+    }
+  }
+  runtime_base_handler_map = {
+    nodejs = "/opt/nodejs/node_modules/datadog-lambda-js/handler.handler"
+    python = "datadog_lambda.handler.handler"
+  }
+  runtime_base_layer_version_map = {
+    nodejs = var.datadog_node_layer_version
+    python = var.datadog_python_layer_version
+  }
+  runtime_layer_map = {
+    "nodejs16.x" = "Datadog-Node16-x"
+    "nodejs18.x" = "Datadog-Node18-x"
+    "nodejs20.x" = "Datadog-Node20-x"
+    "python3.8"  = "Datadog-Python38"
+    "python3.9"  = "Datadog-Python39"
+    "python3.10" = "Datadog-Python310"
+    "python3.11" = "Datadog-Python311"
+    "python3.12" = "Datadog-Python312"
+  }
+}
+
+locals {
+  datadog_lambda_layer_version = lookup(local.runtime_base_layer_version_map, local.runtime_base, null)
+  environment_variables = {
+    common = {
+      DD_CAPTURE_LAMBDA_PAYLOAD  = "false"
+      DD_LOGS_INJECTION          = "false"
+      DD_MERGE_XRAY_TRACES       = "false"
+      DD_SERVERLESS_LOGS_ENABLED = "true"
+      DD_SERVICE                 = var.function_name
+      DD_SITE                    = "datadoghq.com"
+      DD_TRACE_ENABLED           = "true"
+    }
+    runtime = lookup(local.runtime_base_environment_variable_map, local.runtime_base, {})
+  }
+  handler = lookup(local.runtime_base_handler_map, local.runtime_base, var.handler)
+  layers = {
+    extension = ["${local.layer_name_base}:Datadog-Extension${local.layer_suffix}:${var.datadog_extension_layer_version}"]
+    lambda    = local.layer_runtime == null ? [] : ["${local.layer_name_base}:${local.layer_runtime}${local.runtime_base == "nodejs" ? "" : local.layer_suffix}:${local.datadog_lambda_layer_version}"]
+  }
```

Comment:
Isn't there a better way to calculate this value? This doesn't look very readable.

Reply:
I broke out some of this business logic to separate variables to help with the readability